src/write.jl

```diff
@@ -435,7 +435,7 @@ end
 
 function writecell(buf, pos, len, io, x::Integer, opts)
     if x < 0
-        x *= -1
+        x = unsigned(-(x + 1)) + 1
```

Comment:
```suggestion
        x = Base.uabs(x)
```

This should fix the `BigInt` issue.